### DISCUSSION TITLE
Warning if Parameter boundary is reached during fit.

### DISCUSSION THREAD

**Discussion by Julian-Hochhaus** [lmfit/lmfit-py#868] [Q&A] [answered]
Hi,
I wonder if there is some property of `ModelResult` or `MinimizerResult` which holds the information if one of the parameters reached one of its boundaries. 
Reading the docs as well as the code, I was not able to find such a property.  If there truly is no option, what is the best way to check if one of the parameters of a fit result reached a boundary? Iterating all parameters and checking value against min/max value?

Thanks in advance!

Julian

**Comment by reneeotten** [comment]
The information is shown in the fit report, bit there is property that has this information. It should also be fairly straightforward to check yourself: just compare the fitted value to the boundaries you have set. On Apr 24, 2023, at 9:17 AM, Julian-Hochhaus ***@***.***> wrote:﻿
Hi,
I wonder if there is some property of ModelResult or MinimizerResult which holds the information if one of the parameters reached one of its boundaries.
Reading the docs as well as the code, I was not able to find such a property.  If there truly is no option, what is the best way to check if one of the parameters of a fit result reached a boundary? Iterating all parameters and checking value against min/max value?
Thanks in advance!
Julian

**Comment by newville** [comment, marked as answer]
@Julian-Hochhaus @reneeotten we check  with ``np.allcose(par.value, par.min) or np.allcose(par.value, par.max)`` inside of ``fit_report()``.  I could be convinced that we want to add a property for that (and maybe the other thing we also check for ``np.allcose(par.value, par.init_value)``), but it doesn't seem too hard as a downstream check.   

If we have those properties, would we then want to add methods to "lift all boundaries that I hit" or "set vary=False for all parameters at initial values"?  There are probably lots of possibilities there. I doubt we could capture all of them.  In the face of such doubt, simply explaining (maybe FAQ, maybe we need a User Guide) what might be done seems OK.  I lean toward "it's not hard to check and act appropriately downstream".   But, I could be persuaded.

**Reply by Julian-Hochhaus** [reply]
No worries, everything is fine. I was just thinking, since the information was already in the fit_report, if there was some other way to access them that I did not knew about. :)

 By the way, thanks for the heads up on `np.allclose()` - I didn't know about that one!
